core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -134,6 +140,13 @@ public JoinDefinition define(AutoJoinerContext context) {
           case OUTPUT_SCHEMA:
             OutputSchemaError schemaError = (OutputSchemaError) error;
             failure.withOutputSchemaField(schemaError.getField());
+            break;
+          case DISTRIBUTION_SIZE:
+            failure.withConfigProperty(JoinerConfig.DISTRIBUTION_FACTOR);
+            break;
+          case DISTRIBUTION_STAGE:
+            failure.withConfigProperty(JoinerConfig.DISTRIBUTION_STAGE);
+            break;
```

Comment:
this is an existing problem, but we should have a default case to make sure other types of errors (like the generic join error) are collected.

Reply:
The first line before the switch-case handles all the errors:
`ValidationFailure failure = collector.addFailure(error.getMessage(), error.getCorrectiveAction());`
Its just that the errors with types get modified to target a particular property. So the generic errors are captured, they just appear at the top of the plugin instead of underneath a given property

Reply:
ah ok, makes sense